database/errors.go

```diff
@@ -2,12 +2,15 @@ package database
 
 import (
 	"errors"
+	"fmt"
 )
 
-var (
-	// ErrTableNotFound is returned when the targeted table doesn't exist.
-	ErrTableNotFound = errors.New("table not found")
```

Comment:
I’d suggest to keep the error variable and call `fmt.Errorf("%w: %q", ErrTableNotFound, tableName)` at each fail point.

Then you’d need to replace all `err == ErrTableNotFound ` comparisons with `errors.Is(err, ErrTableNotFound)`.

Reply:
Very good idea!

Reply:
Thanks for the answer, I'm going to follow your suggestions :)